index.bs

```diff
@@ -353,13 +370,17 @@ The <dfn method for=FileSystemFileHandle>createWritable(|options|)</dfn> method
   1. If |access| is not "{{PermissionState/granted}}",
      [=reject=] |result| with a "{{NotAllowedError}}" {{DOMException}} and abort.
   1. Let |entry| be [=this=]'s [=FileSystemHandle/entry=].
-  1. Let |lockResult| be the result of [=file entry/lock/take|taking a lock=] with "`shared`" on |entry|.
-  1. If |lockResult| is false, [=reject=] |result| with a "{{NoModificationAllowedError}}" {{DOMException}} and abort.
-  1. Let |stream| be the result of [=create a new FileSystemWritableFileStream|creating a new FileSystemWritableFileStream=]
-     for |entry| in [=this=]'s [=relevant realm=].
-  1. If |options|.{{FileSystemCreateWritableOptions/keepExistingData}} is true:
-    1. Set |stream|.[=[[buffer]]=] to a copy of |entry|'s [=file entry/binary data=].
-  1. [=/Resolve=] |result| with |stream|.
+  1. Let |lockResultAlgorithm| be an algorithm which takes a |lockResult| and
+     runs these steps:
+    1. If |lockResult| is "`failure`", [=reject=] |result| with a
+       "{{NoModificationAllowedError}}" {{DOMException}} and abort.
+    1. Let |stream| be the result of [=create a new FileSystemWritableFileStream|creating a new FileSystemWritableFileStream=]
```

Comment:
This really needs to happen from a task. Same below.

Reply:
Also now happens on a task on the `file system queue`